### ISSUE TITLE
Wrong Props while using ForwardRef

### ISSUE THREAD

**Issue by dkirchhof** [Zaid-Ajaj/Feliz#392]
Hey,

since a few days i'm experimenting with this great library. But now i have a problem while forwarding refs.  
Here are two example components to reproduce the issue:

```fs
type TestProps = 
    {
        defaultValue: string
    }

let testComponent =
    React.forwardRef
        (fun (props: TestProps, ref) ->
            Browser.Dom.console.log props
            Browser.Dom.console.log ref 

            Html.input [
                prop.ref ref
                prop.value props.defaultValue
            ]
        )

let someOtherComponent () =
    let inputRef = React.useInputRef()

    Html.div [
        test ({ defaultValue = "test" }, inputRef)
    ]
```

I had some undefined here and undefined there errors. So i added the `console.log`.  

![image](https://user-images.githubusercontent.com/2778643/128472114-030f3108-9720-418c-9622-8f494b6286d4.png)

Well, as you can see, the props object will be nested. To get the `defaultValue` i have to type `props.props.defaultValue`, but this will produce some type errors.

**Comment by Zaid-Ajaj** [comment]
Hi there @dkirchhof thanks for filing the issue, I will need to look into it soon. I am bit low on time these days, if you can help me out with a proposal on how to fix the problem, it would be awesome 💯 to be honest, I haven't worked with the ForwardRef API a lot so I will some time to dig into it

**Comment by dkirchhof** [comment]
I will try when i have some time left :)

**Comment by dkirchhof** [comment]
Okay, i made a PR and the build process was successfull. Unfortunately i wasn't able to build feliz locally, so i haven't tested it :crossed_fingers:

**Comment by Zaid-Ajaj** [comment]
Thanks a lot for the fix @dkirchhof 🙏 it looks good to me. Merged and published as of Feliz v1.51 🚀 can you give it a try and see if it resolves the problem?

**Comment by dkirchhof** [comment]
Hey, i can confirm the the bug is fixed now :slightly_smiling_face:
